llvm/lib/Target/DPU/DPURegisterInfo.cpp

```diff
@@ -106,8 +133,21 @@ void DPURegisterInfo::eliminateFrameIndex(MachineBasicBlock::iterator II,
   default: {
     Offset += MI.getOperand(FIOperandNum + 1).getImm();
 
-    MI.getOperand(FIOperandNum).ChangeToRegister(FrameReg, false);
-    MI.getOperand(FIOperandNum + 1).ChangeToImmediate(Offset);
+    unsigned StoreRegInst;
+    if (isStoreImmInst(Opcode, StoreRegInst) && !isInt<12>(Offset)) {
```

Comment:
I would inverse the tests because most of the time int are inferior than `2^12`

Reply:
Done